src/CodeFormatter.cs

```diff
@@ -293,99 +289,105 @@ private static void LogWorkspaceDiagnostics(ILogger logger, bool logWorkspaceWar
 
-            var addedFilePaths = new HashSet<string>(documentsAndOptions.Length);
-            var formattableFiles = ImmutableArray.CreateBuilder<(DocumentId, OptionSet, ICodingConventionsSnapshot)>(documentsAndOptions.Length);
-            foreach (var (document, options, codingConventions, hasEditorConfig) in documentsAndOptions)
-            {
-                if (document?.FilePath is null)
-                {
-                    continue;
-                }
+                    if (!fileMatcher.Match(document.FilePath).HasMatches ||
+                        !document.SupportsSyntaxTree)
+                    {
+                        continue;
+                    }
 
-                // If any code file has an .editorconfig, then we should ignore files without an .editorconfig entry.
-                if (foundEditorConfig && !hasEditorConfig)
-                {
-                    continue;
-                }
+                    var syntaxTree = await document.GetSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+                    if (syntaxTree is null)
+                    {
+                        throw new Exception($"Unable to get a syntax tree for '{document.Name}'");
+                    }
 
-                // If we've already added this document, either via a link or multi-targeted framework, then ignore.
-                if (addedFilePaths.Contains(document.FilePath))
-                {
-                    continue;
+                    if (!includeGeneratedFiles &&
+                        await GeneratedCodeUtilities.IsGeneratedCodeAsync(syntaxTree, cancellationToken).ConfigureAwait(false))
+                    {
+                        continue;
+                    }
+
+                    var analyzerConfigOptions = await GetAnalyzerConfigOptionsAsync(document, cancellationToken).ConfigureAwait(false);
+                    var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+
+                    var formattableDocument = new DocumentWithOptions(document, optionSet, analyzerConfigOptions);
+
+                    // Track files covered by and editorconfig and those not covered.
+                    if (formattableDocument.AnalyzerConfigOptions is object)
+                    {
+                        documentsCoveredByEditorConfig.Add(formattableDocument);
+                    }
+                    else
+                    {
+                        documentsNotCoveredByEditorConfig.Add(formattableDocument);
+                    }
                 }
-
-                addedFilePaths.Add(document.FilePath);
-                formattableFiles.Add((document.Id, options, codingConventions));
             }
 
-            return (fileCount, formattableFiles.ToImmutableArray());
+            // If no files are covered by an editorconfig, then return them all. Otherwise only return
+            // files that are covered by an editorconfig.
```

Comment:
I admit I'm not understanding why here -- can the comment clarify?

Reply:
I will add one. Initially we would format all document, even if the file wasn't covered by and .editorconfig and would have defaults applied. This behavior was an early requested change since users were surprised to have files not specified by the .editorconfig modified. My assumption is that users without an .editorconfig still wanted formatting (they did run a formatter after all), so it runs on all files with defaults.